plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreReadAccessOperations.java

```diff
@@ -0,0 +1,340 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.thrift;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultiset;
+import io.trino.plugin.hive.HiveType;
+import io.trino.plugin.hive.TestingHivePlugin;
+import io.trino.plugin.hive.metastore.Column;
+import io.trino.plugin.hive.metastore.CountingAccessHiveMetastore;
+import io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods;
+import io.trino.plugin.hive.metastore.CountingAccessHiveMetastoreUtil;
+import io.trino.plugin.hive.metastore.Database;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import io.trino.testng.services.Flaky;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.connector.informationschema.InformationSchemaMetadata.MAX_PREFIXES_COUNT;
+import static io.trino.plugin.hive.HiveStorageFormat.ORC;
+import static io.trino.plugin.hive.TableType.MANAGED_TABLE;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_ALL_DATABASES;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_ALL_TABLES_WITH_PARAMETER;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_ALL_VIEWS_WITH_PARAMETER;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_TABLE;
+import static io.trino.plugin.hive.metastore.PrincipalPrivileges.NO_PRIVILEGES;
+import static io.trino.plugin.hive.metastore.StorageFormat.fromHiveStorageFormat;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+@Test(singleThreaded = true) // metastore invocation counters shares mutable state so can't be run from many threads simultaneously
+public class TestHiveMetastoreReadAccessOperations
```

Comment:
`TestHiveMetastoreMetadataQueriesAccessOperations`

BTW 
We discussed to keep this separate from `TestHiveMetastoreAccessOperations` to avoid test interaction issues
however the test is single-threaded anyway, so all we need to do is add cleanup (drop table) in test methods.
this would avoid having two similar test classes.

Reply:
Creating those schemas and tables took 16-18 seconds on my laptop, so doing this for each test method is not convenient

Reply:
> `TestHiveMetastoreMetadataQueriesAccessOperations`

please rename


> Creating those schemas and tables took 16-18 seconds on my laptop, so doing this for each test method is not convenient

this is done once per class, not per test method

Reply:
Renamed